### ISSUE TITLE
Conflict with cordova-plugin-firebase-dynamiclinks on android

### ISSUE THREAD

**Issue by nitrique** [microsoft/cordova-plugin-code-push#600]
### Description

I added codepush on android with cordova-plugin-firebase-dynamiclinks already installed, I got on graddle sync step :

```
ERROR: In project 'app' a resolved Google Play services library dependency depends on another at an exact version (e.g. "[1.3.1
,2.3]", but isn't being resolved to that version. Behavior exhibited by the library will be unknown.

Dependency failing: com.nimbusds:nimbus-jose-jwt:5.1 -> net.minidev:json-smart@[1.3.1,2.3], but json-smart version was 2
.3.

The following dependencies are project dependencies that are direct or have transitive dependencies that lead to the art
ifact with the issue.
-- Project 'app' depends onto net.minidev:json-smart@{strictly 2.3}
-- Project 'app' depends onto com.nimbusds:nimbus-jose-jwt@{strictly 5.1}
-- Project 'app' depends onto com.nimbusds:nimbus-jose-jwt@5.1

For extended debugging info execute Gradle from the command line with ./gradlew --info :app:assembleDebug to see the dep
endency paths to the artifact. This error message came from the google-services Gradle plugin, report issues at https://
github.com/google/play-services-plugins and disable by adding "googleServices { disableVersionCheck = false }" to your b
uild.gradle file.
```

### Reproduction

Start a cordova project

- Add platform android
- Install cordova-plugin-firebase-dynamiclinks
- Install cordova-plugin-code-push

Build for Android

### Additional Information

* cordova-plugin-code-push version:  1.12.0 
* List of installed plugins:
```
code-push 3.0.1 "CodePushAcquisition"
cordova-plugin-apprate 1.5.0 "AppRate"
cordova-plugin-browsertab 0.2.0 "cordova-plugin-browsertab"
cordova-plugin-code-push 1.12.0 "CodePush"
cordova-plugin-compat 1.2.0 "Compat"
cordova-plugin-device 1.1.7 "Device"
cordova-plugin-dialogs 2.0.2 "Notification"
cordova-plugin-file 4.3.3 "File"
cordova-plugin-file-transfer 1.6.3 "File Transfer"
cordova-plugin-firebase-dynamiclinks 4.0.3 "FirebaseDynamicLinksPlugin"
cordova-plugin-ionic-keyboard 2.2.0 "cordova-plugin-ionic-keyboard"
cordova-plugin-ionic-webview 4.1.3 "cordova-plugin-ionic-webview"
cordova-plugin-ios-non-exempt-encryption 1.0.0 "ITSAppUsesNonExemptEncryption false"
cordova-plugin-nativestorage 2.3.2 "NativeStorage"
cordova-plugin-splashscreen 5.0.3 "Splashscreen"
cordova-plugin-statusbar 2.4.3 "StatusBar"
cordova-plugin-whitelist 1.3.4 "Whitelist"
cordova-plugin-zip 3.1.0 "cordova-plugin-zip"
cordova-support-android-plugin 1.0.1 "cordova-support-android-plugin"
cordova-support-google-services 1.3.2 "cordova-support-google-services"
```
* Cordova version: 9.0.0 (cordova-lib@9.0.1)
* iOS/Android/Windows version: android 8.1.0
* Does this reproduce on a debug build or release build? any build
* Does this reproduce on a simulator, or only on a physical device? any environement

Thanks for support,
Regards.
Nicolas

**Comment by nitrique** [comment]
ping

**Comment by brassier** [comment]
I ran into this with the FirebaseX plugin (conflicting with CodePush).  I ran into 2 different solution options:
1) Adjust the CodePush build-extras.gradle file to exclude part of the nimbus-jose-jwt package that seems to be included in other Firebase/Google service packages.  Explained [here](https://github.com/arnesson/cordova-plugin-firebase/issues/1086#issuecomment-504353591).
2) Adjust the Firebase plugin to upgrade to google-services 4.3.3.  The conflict seems to be handled better with that.  Explained [here](https://github.com/microsoft/react-native-code-push/issues/1448#issuecomment-561130291).  In your case it looks like this may be needed in the [cordova-support-google-services](https://github.com/chemerisuk/cordova-support-google-services/blob/f9d05e7ddc0b738e658019f2cf429190f5b05ef5/build.gradle#L10) plugin (which is pulled in with the [firebase-dynamiclinks](https://github.com/chemerisuk/cordova-plugin-firebase-dynamiclinks/blob/19434ea7a3d40748bd7a900bfa704531cbd9b67e/plugin.xml#L55) plugin).
